### ISSUE TITLE
Error: invalid pattern - glob in CommonJS doesn't seem to accept arrays of patterns like glob in ESM does

### ISSUE THREAD

**Issue by jamesmortensen** [isaacs/node-glob#532]
Working on trying to understand how glob and globSync works in a CommonJS module.  When I attempt to follow the docs to require globSync, I get back `undefined`.  With ESM, everything works great, so I attempted to load glob using a dynamic import:

```
const glob = (await import('glob')).default;
```

The object I get back contains the following methods:

```
glob
Glob
GlobSync
hasMagic
sync
```

I have attempted to use glob.sync, but it only allows simple pattern matchers as strings, whereas the glob I import in ESM modules works with an array of matchers.  I get an error inside Minimatch because the pattern is an array instead of a string, whereas glob in ESM accepts arrays of strings.

Even with dynamic import, like so:

```
const glob = (await import('glob')).default.glob;

const fileMatchers = [
        './test/**/*.spec.js',
        './test/**/*.test.js',
        './test/**/*.spec.cjs',
        './test/**/*.test.cjs',
        './test/**/*.spec.mjs',
        './test/**/*.test.mjs',
    ];

// this line throws "invalid pattern" because fileMatchers is not a string. Only a problem when using require or dynamic import.
const testFiles = glob(fileMatchers, { ignore: 'node_modules/**' });
```

I see the same error even when requiring glob as per the docs:

```javascript
async function getTestFiles() {
  const { glob } = require('glob'); 

  const fileMatchers = [
        './test/**/*.spec.js',
        './test/**/*.test.js',
        './test/**/*.spec.cjs',
        './test/**/*.test.cjs',
        './test/**/*.spec.mjs',
        './test/**/*.test.mjs',
    ];

  const testFiles = await glob(fileMatchers, { ignore: 'node_modules/**' });
  return testFiles;
}
```

**Error message:**
```
de_modules/minimatch/minimatch.js:275
    throw new TypeError('invalid pattern')
```

When I trace the code with the debugger, I don't find minimatch used when running in ESM modules, only CommonJS.  Not sure why.  Hope this information is helpful.  

As an aside, the docs mention using `const { globSync } = require('glob');` to get the sync version, but it returns undefined.

Please let me know if I can provide any further details.  Thank you for your work on this module!

**Comment by jamesmortensen** [comment]
After further investigation, I discovered the problem was that I was using an older version of glob in the project, but a newer version inside a module I wrote that also used glob.  The module was using the latest version, 10.2.7, but the main application was using the glob which was loaded by mocha, v7.2.0.

After installing glob in the project dependencies, everything worked as expected.

Thus, there is no issue here.  Hope this ticket somehow helps others!
